packages/node_modules/@node-red/editor-client/src/js/ui/common/treeList.js

```diff
@@ -348,6 +348,16 @@
                 that._selected.delete(item);
                 delete item.treeList;
                 delete that._items[item.id];
+                if(item.depth === 0) {
+                    for(var key in that._items) {
+                        var child = that._items[key];
+                        if(child.parent && child.parent.id === item.id) {
+                            delete that._items[key].treeList;
+                            delete that._items[key];
+                        }
+                    }
+                    that._data = that._data.filter(data => data.id !== item.id)
```

Comment:
We do not use arrow functions in the editor due to requirements to support IE11.

```suggestion
                    that._data = that._data.filter(function(data) { return data.id !== item.id})
```